### PR TITLE
Refactor czi_hosted and server into backend directory, pull common code into backend/common, refactor tests 

### PR DESCRIPTION
**Pull request by MDunitz**
#### Reviewers
This touches all of the cellxgene backend code so I think most of the team should take a look

---

## Changes
Refactor cellxgene backend code
- backend
    - server (previously local_server)
    - czi_hosted (previously server)
    - common (code common to both)
    - test
        - unit
            - test_common
            - test_czi_hosted
            - test_server
        - fixtures
        
        
Code common to both services include
- fbs
- utils
    - type_conversion
    - utils
    - aws_secrets
    - data_locater
- colors
- constants
- errors

This pr also updates the compatibility tests to use the server (previously local_server) instead of czi_hosted (previously server) because we pin versions for the czi_hosted backend but the compatibility tests are intended for people using the CLI who may have varying versions of anndata installed.
There is some remaining code that is shared between both server and czi_hosted. 
The scanpy and diffexp code will be removed from one of the services in the near future.
I think keeping the config code separate will allow us to significantly simplify it for both services but I'm curious if other people think otherwise.

### CODE REVIEW

File: .dockerignore
Comment (maniarathi) [inline .dockerignore:6]: Can we remove this?

File: .github/workflows/compatibility_tests.yml
Comment (maniarathi) [inline .github/workflows/compatibility_tests.yml:44]: I think the compatibility tests should probably be testing against the desktop version, not the czi-hosted backend. This is because we pin specific versions for the czi-hosted backend but the compatibility tests are meant to be for folks using the CLI who may have varying versions of anndata installed.

File: backend/czi_hosted/eb/app.py
Comment (maniarathi) [inline backend/czi_hosted/eb/app.py:195]: Why the change to True?

File: backend/czi_hosted/test/__init__.py
Comment (maniarathi) [inline backend/czi_hosted/test/__init__.py:1]: Fully commented out file?

File: backend/server/test/__init__.py
Comment (maniarathi) [inline backend/server/test/__init__.py:1]: Fully commented out file?

File: backend/server/test/fixtures/tmp_dir/environment_external_config.yaml
Comment (maniarathi) [inline backend/server/test/fixtures/tmp_dir/environment_external_config.yaml:1]: Are the files in `tmp_dir` intentional?

File: Makefile
Comment (maniarathi) [inline Makefile:213]: Were the changes to pip3 intentional? We should probably be uniform as to whether `pip` or `pip3` across this Makefile.

File: backend/czi_hosted/auth/__init__.py
Comment (maniarathi) [inline backend/czi_hosted/auth/__init__.py:3]: My guess is that the lack of `import backend.czi_hosted_auth.auth_none` in this file is what is causing your czi-hosted unit test to fail currently.

File: .dockerignore
Comment (MDunitz) [reply .dockerignore:6]: The file? or the line?

File: .github/workflows/compatibility_tests.yml
Comment (MDunitz) [reply .github/workflows/compatibility_tests.yml:44]: doh! updated

File: backend/czi_hosted/eb/app.py
Comment (MDunitz) [reply backend/czi_hosted/eb/app.py:195]: just for debugging -- removed

File: backend/server/test/fixtures/tmp_dir/environment_external_config.yaml
Comment (MDunitz) [reply backend/server/test/fixtures/tmp_dir/environment_external_config.yaml:1]: no -- removed

File: Makefile
Comment (MDunitz) [reply Makefile:213]: I'll revert back to pip

File: backend/server/test/fixtures/tmp_dir/environment_external_config.yaml
Comment (MDunitz) [reply backend/server/test/fixtures/tmp_dir/environment_external_config.yaml]: no -- removed

File: .dockerignore
Comment (bkmartinjr) [reply .dockerignore:6]: the line.  not sure why the file specifies a directory that no longer exists....

File: Makefile
Comment (bkmartinjr) [inline Makefile:5]: I think it would be clearer if the variables reflected their content, eg, `s/SERVERBUILD/CZIHOSTEDBUILD/g` and `s/LOCALSERVERBUILD/SERVERBUILD/g`.

At your discretion.  But it caused a small bit of confusion for me.

File: backend/common/fbs/NetEncoding/Column.py
Comment (bkmartinjr) [inline backend/common/fbs/NetEncoding/Column.py:7]: FYI only:   these files in `NetEncoding` are machine-generated.  While there is no harm in linting / prettying them, there is also no benefit, and we'll lose the fixes as soon as we regenerate the files (not commonly done).

File: backend/czi_hosted/eb/app.py
Comment (bkmartinjr) [reply backend/czi_hosted/eb/app.py:188]: it seems to still be `True`?

File: backend/czi_hosted/test/performance/scale_test_annotations.py
Comment (bkmartinjr) [inline backend/czi_hosted/test/performance/scale_test_annotations.py:10]: shouldn't this be `backend/czi_hosted/...`?   (ditto for the same string on line 13)

File: backend/test/fixtures/database/__init__.py
Comment (bkmartinjr) [inline backend/test/fixtures/database/__init__.py:7]: shouldn't this file be in `backend/czi_hosted/test/fixtures/...`?

**Review by bkmartinjr — CHANGES_REQUESTED** [review]
Overall looks good.  A few small issues noted, plus tests need to run.  Please re-tag me when ready for a second pass review.

**Review by maniarathi — APPROVED** [review]
Please wait for Bruce's LGTM as well, but after resolving the remaining open comments, LGTM from me.

File: Makefile
Comment (MDunitz) [reply Makefile:5]: updated

File: backend/test/fixtures/database/__init__.py
Comment (MDunitz) [reply backend/test/fixtures/database/__init__.py:7]: Moved all of the tests/infrastructure into test directory

File: backend/test/test_common/__init__.py
Comment (bkmartinjr) [inline backend/test/test_common/__init__.py:1]: any reason this isn't an empty file?

Comment (MDunitz) [reply backend/test/test_common/__init__.py]: no --removed

File: backend/czi_hosted/eb/app.py
Comment (bkmartinjr) [reply backend/czi_hosted/eb/app.py:188]: still not resolved!

**Review by bkmartinjr — APPROVED** [review]
There is one small issue remaining in `backend/czi_hosted/eb/app.py` . Otherwise, it looks good to me.  Approving.

File: backend/czi_hosted/eb/app.py
Comment (MDunitz) [reply backend/czi_hosted/eb/app.py]: Sorry about that!

**Pull request merged**
